### PR TITLE
[Backport-v2.2-branch] bluetooth: host: fix wrong bt/cf settings loading

### PR DESCRIPTION
**Pull request by fnde-ot**
This commits fixes the loading of bt/cf settings into memory. Only data
was loaded and not the address.

Backport of #25543

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

**Comment by fnde-ot** [comment]
Poking @jhedberg (release manager for 2.2)

**Comment by joerchan** [comment]
Maybe remove the fixes links? Otherwise merging the backport before the original has been merged would close the issue before it is in master.

**Comment by fnde-ot** [comment]
> Maybe remove the fixes links? Otherwise merging the backport before the original has been merged would close the issue before it is in master.

Done!

**Review by joerchan — APPROVED** [review]

**Pull request merged**
